File: problems/132_pattern/solution.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) {
        int n=nums.size();
        stack<int>st;

        int se=-1e9;
        for(int i=n-1;i>=0;i--){
            if(nums[i]<se)
            return 1;
            else{
                while(!st.empty() && st.top()<nums[i])
                se=st.top(),st.pop();
            }
            st.push(nums[i]);
        }
        return 0;
    }
};

File: problems/132_pattern/solution.go
func find132pattern(A []int) bool {
    var stack []int
    second:=math.MinInt32
    for i:=len(A)-1;i>=0;i-- {
        if A[i]<second {
            return true
        }else{
            for len(stack)>0 && stack[len(stack)-1]<A[i] {
                second=stack[len(stack)-1]
                stack=stack[:len(stack)-1]
            }
        }
        stack=append(stack,A[i])
    }
    return false
}